File: 917-boats-to-save-people/boats-to-save-people.py
class Solution:
    def numRescueBoats(self, people: List[int], limit: int) -> int:
        people.sort()
        l, r = 0, len(people) - 1
        count = 0
        while l <= r:
            wei = people[l] + people[r]
            if wei <= limit:
                l += 1
            r -= 1
            count += 1
        return count

File: 917-boats-to-save-people/boats-to-save-people.cpp
class Solution {
public:
    int numRescueBoats(vector<int>& people, int limit) {
        std::sort(people.begin(), people.end());
        int l = 0;
        int r = people.size() - 1;
        int count = 0;
        while (l <= r) {
            int weight = people[l] + people[r];
            if (weight <= limit) {
                l += 1;
            }
            r -= 1;
            count += 1;
        }
        return count;
    }
};